pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -246,12 +246,12 @@ public void run() {
 
     private void loadJars() throws Exception {
 
-        if (jarFile.endsWith(".nar")) {
-            // The functions code is contained in a NAR archive
+        try {
+            // Let's first try to treat it as a nar archive
             fnCache.registerFunctionInstanceWithArchive(instanceConfig.getFunctionId(), instanceConfig.getInstanceId(),
                     jarFile);
-        } else {
-            log.info("Loading JAR files for function {} from archive {}", instanceConfig, jarFile);
+        } catch (Exception e) {
```

Comment:
we should probably just catch the exception that we expect registerFunctionInstanceWithArchive to throw if its a jar and not a blanket catch

Reply:
Changed